src/andromede/simulation/benders_decomposed.py

```diff
@@ -125,62 +143,70 @@ def export_options(
+            serialize(
+                f"{subproblem.name}.mps", subproblem.export_as_mps(), self.emplacement
+            )
+        serialize("structure.txt", self.export_structure(), self.emplacement)
+        serialize_json(
             "options.json",
-            json.dumps(
-                self.export_options(solver_name=solver_name, log_level=log_level),
-                indent=4,
-            ),
-            directory,
+            self.export_options(solver_name=solver_name, log_level=log_level),
+            self.emplacement,
         )
 
         if is_debug:
-            serialize("master.lp", self.master.export_as_lp(), directory)
-            serialize("subproblem.lp", self.subproblems[0].export_as_lp(), directory)
+            serialize("master.lp", self.master.export_as_lp(), self.emplacement)
+            for subproblem in self.subproblems:
+                serialize(
+                    f"{subproblem.name}.lp", subproblem.export_as_lp(), self.emplacement
+                )
+
+    def read_solution(self) -> None:
+        try:
+            data = read_json("out.json", self.emplacement / self.output_path)
+
+        except FileNotFoundError:
+            # TODO For now, it will return as if nothing is wrong
+            # modify it with runner's run
+            print("Return without reading it for now")
+            return
+
+        if self.is_merged:
+            self.solution = BendersMergedSolution(data)
+        else:
+            self.solution = BendersDecomposedSolution(data)
 
     def run(
         self,
         *,
-        path: str = "outputs/lp",
         solver_name: str = "XPRESS",
         log_level: int = 0,
+        should_merge: bool = False,
     ) -> bool:
-        self.prepare(path=path, solver_name=solver_name, log_level=log_level)
-        root_dir = pathlib.Path().cwd()
-        path_to_benders = root_dir / "bin" / "benders"
-
-        if not path_to_benders.is_file():
-            # TODO Maybe a more robust check and/or return value?
-            # For now, it won't look anywhere else because a new
-            # architecture should be discussed
-            print(f"{path_to_benders} executable not found. Returning True")
-            return True
+        self.prepare(solver_name=solver_name, log_level=log_level)
 
-        os.chdir(path)
-        res = subprocess.run(
-            [path_to_benders, "options.json"],
-            stdout=sys.stdout,
-            stderr=subprocess.DEVNULL,  # TODO For now, to avoid the "Invalid MIT-MAGIC-COOKIE-1 key" error
-            shell=False,
-        )
-        os.chdir(root_dir)
+        if not should_merge:
+            return_code = BendersRunner(self.emplacement).run()
+        else:
+            self.is_merged = True
+            return_code = MergeMPSRunner(self.emplacement).run()
 
-        return res.returncode == 0
+        if return_code == 0:
+            self.read_solution()
+            return True
+        else:
+            return False
```

Comment:
Raise an error message ?

Reply:
I agree. For now I would leave like this and would raise an error in the PR with the CI integration